sklearn/compose/_column_transformer.py

```diff
@@ -1652,3 +1676,13 @@ def _with_dtype_warning_enabled_set_to(warning_enabled, transformers):
             )
         result.append((name, trans, columns))
     return result
+
+
+def _feature_names_out(
```

Comment:
There are many `_feature_names_out` from the call-site. I think we should give this an more explicit name:

```suggestion
def _feature_names_out_with_str_format(
```

Reply:
I'm not a fan. Creating new functions/methods actually hides functionalities away from the user. Indeed, they will often look for API reference for one particular function and check its parameters. However, I feel like it is rare that users will look at alternative functions.

In this case `_feature_names_out` is not bloated at all. The additional parameter makes it easy for the user to know that using string format is possible. I don't think it is an issue that there are many _feature_names_out from the call-site.

Reply:
edit: nvm, it as been a long time since I've checked this PR. I forgot that `_feature_names_out` is a private function. I'll see what I can do :)